azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_ops_iothub.py

```diff
@@ -73,29 +73,36 @@ def __init__(
         module_id=None,
         gateway_hostname=None,
         ca_cert=None,
+        client_cert=None,
+        sas_token=None,
         callback=None,
     ):
         """
-        Initializer for SetAuthProviderArgsOperation objects.
+        Initializer for SetIoTHubConnectionArgsOperation objects.
 
         :param str device_id: The device id for the device that we are connecting.
         :param str hostname: The hostname of the iothub service we are connecting to.
         :param str module_id: (optional) If we are connecting as a module, this contains the module id
           for the module we are connecting.
         :param str gateway_hostname: (optional) If we are going through a gateway host, this is the
           hostname for the gateway
-        :param str ca_cert: (Optional) The CA certificate to use if theMQTT server that we're going to
+        :param str ca_cert: (Optional) The CA certificate to use if the server that we're going to
           connect to uses server-side TLS
+        :param X509 client_cert: (Optional) The x509 object containing a client certificate and key used to connect
+          to the service
+        :param str sas_token: The token string which will be used to authenticate with the service
```

Comment:
>sas_token [](start = 19, length = 9)

optional ? may be we can say that authentication of the client should be done either via sas or cert. so one of them should be present.